src/plugins/dashboard/public/dashboard_app/top_nav/add_new_panel/dashboard_panel_selection_flyout.tsx

```diff
@@ -0,0 +1,285 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React, { useEffect, useState } from 'react';
+import { i18n as i18nFn } from '@kbn/i18n';
+import { type EuiFlyoutProps, EuiLoadingChart } from '@elastic/eui';
+import { type Observable } from 'rxjs';
+import orderBy from 'lodash/orderBy';
+import {
+  EuiEmptyPrompt,
+  EuiButtonEmpty,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiFlyoutBody,
+  EuiFlyoutFooter,
+  EuiFlyoutHeader,
+  EuiForm,
+  EuiBadge,
+  EuiFormRow,
+  EuiTitle,
+  EuiFieldSearch,
+  useEuiTheme,
+  EuiListGroup,
+  EuiListGroupItem,
+  EuiToolTip,
+  EuiText,
+} from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n-react';
+import {
+  type PanelSelectionMenuItem,
+  type GroupedAddPanelActions,
+} from './add_panel_action_menu_items';
+
+export interface DashboardPanelSelectionListFlyoutProps {
+  /** Handler to close flyout */
+  close: () => void;
+  /** Padding for flyout  */
+  paddingSize: Exclude<EuiFlyoutProps['paddingSize'], 'none' | undefined>;
+  /** Handler for creating new visualization of a specified type */
+  dashboardPanels$: Observable<GroupedAddPanelActions[]>;
+}
+
+export const DashboardPanelSelectionListFlyout: React.FC<
+  DashboardPanelSelectionListFlyoutProps
+> = ({ paddingSize, close, dashboardPanels$ }) => {
+  const { euiTheme } = useEuiTheme();
+  const [{ data: panels, loading, error }, setPanelState] = useState<{
+    loading: boolean;
+    data: GroupedAddPanelActions[] | null;
+    error: unknown | null;
+  }>({ loading: true, data: null, error: null });
+
+  const [searchTerm, setSearchTerm] = useState<string>('');
+  const [panelsSearchResult, setPanelsSearchResult] = useState<GroupedAddPanelActions[] | null>(
+    panels
+  );
+
+  useEffect(() => {
+    if (loading) {
+      const subscription = dashboardPanels$.subscribe({
```

Comment:
I wonder if using an observable here is the most straightforward way to get an async list of panel types into this flyout? IMO, an observable should be used when we expect the state to change more than once over the lifetime of the component.

When we have a list of panel types that is populated once and not expected to change it might be better to use a `Promise` instead,  and use the same `useEffect` approach to await the results. Then if the user opens the flyout again and the list is cached, you could pass in `Promise.resolve(cachedPanelTypes)` instead.

Reply:
Fair enough... no second thought there's actually no reason exposing this particular implementation detail to the component. I've made modifications so we only pass an async function in place of an observable